hudi-client/hudi-client-common/src/main/java/org/apache/hudi/client/BaseHoodieWriteClient.java

```diff
@@ -839,6 +842,12 @@ private void startCommitWithTime(String instantTime, String actionType, HoodieTa
 
   private void startCommit(String instantTime, String actionType, HoodieTableMetaClient metaClient) {
     LOG.info("Generate a new instant time: " + instantTime + " action: " + actionType);
+    // check there are no inflight restore before starting a new commit.
+    ValidationUtils.checkArgument(metaClient.getActiveTimeline().getRestoreTimeline().filterInflightsAndRequested().countInstants() == 0,
+        "Found pending restore in active timeline. Please complete the restore fully before proceeding. As of now, table could be in an inconsistent state. "
+          + "Pending restores: " + Arrays.toString(metaClient.getActiveTimeline().getRestoreTimeline()
+          .filterInflightsAndRequested().getInstantsAsStream().map(instant -> instant.getTimestamp()).collect(Collectors.toList()).toArray()));
```

Comment:
`metaClient.getActiveTimeline().getRestoreTimeline().filterInflightsAndRequested()` avoid invoking this twice?